### PR TITLE
Add OpenStack image build upload procedure

### PR DESCRIPTION
**Pull request by dkistner**
**What this PR does / why we need it**:
Same as https://github.com/gardenlinux/gardenlinux/pull/91 to different traget branch `main`.

**Release note**:

```improvement operator
The OpenStack image upload procedure is now integrated.
```

### CODE REVIEW

File: ci/glci/model.py [hunk 1]
```diff
@@ -316,7 +323,8 @@ class ReleaseManifest(ReleaseIdentifier):
     build_timestamp: str
     paths: typing.Tuple[typing.Union[S3_ReleaseFile]]
     published_image_metadata: typing.Union[AlicloudPublishedImageSet,
-                                           AwsPublishedImageSet, AzurePublishedImage, GcpPublishedImage, None]
+                                           AwsPublishedImageSet, AzurePublishedImage,
```
Comment (ccwienk) [inline ci/glci/model.py:326]: as discussed yesterday, I recommend formatting list member one entry per line and order alphabetically

File: ci/glci/model.py [hunk 2]
```diff
@@ -465,9 +473,26 @@ class AzurePublishCfg:
...
+    region: str
+    project_name: str
+    username: str
+    password: str
+
+
+@dataclasses.dataclass(frozen=True)
+class OpenstackPublishCfg:
+    openrc: OpenstackEnviroment
+    properties: dict
```
Comment (ccwienk) [inline ci/glci/model.py:489]: while this technically works, of course, adding dataclasses also for the `properties` could help to improve readability / maintainability

File: ci/glci/model.py (hunk 1 above)
Comment (msohn) [reply ci/glci/model.py:326]: you don't use an auto-formatter like black [1] ?

[1] https://github.com/psf/black

File: ci/glci/model.py (hunk 2 above)
Comment (dkistner) [reply ci/glci/model.py:489]: Hmm the issue here we need the properties dict to pass image properties which can be different depending on the openstack environment. As everything within the ci comes from the secret server we should maybe also retrieve them from there. WDYT?

**Comment by dkistner** [comment]
I have updated the PR. The openstack gl ci use now the secret server as input.
This PR require: https://github.com/gardener/cc-utils/pull/446
After this is merged I will remove the draft status.

File: ci/glci/model.py (hunk 2 above)
Comment (dkistner) [reply ci/glci/model.py]: The information are now retrieved from secret server.

**Pull request merged**